src/ArmTemplates/Extractor/Utilities/DataProcessors/GroupDataProcessor.cs

```diff
@@ -0,0 +1,64 @@
+﻿// --------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+// --------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Templates.Groups;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.Models;
+using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.Utilities.DataProcessors.Absctraction;
+
+namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Extractor.Utilities.DataProcessors
+{
+    public class GroupDataProcessor: IGroupDataProcessor
+    {
+        public IDictionary<string, string> OverrideRules {
+            get => this.overrideRules;  
+            set => this.overrideRules = value;
+        }
+
+        IDictionary<string, string> overrideRules;
```

Comment:
There is no need in this field. 
![image](https://user-images.githubusercontent.com/31598696/168560638-a6a8cd5f-7c45-46a6-a024-585e62df87dd.png)
decompiles to 
![image](https://user-images.githubusercontent.com/31598696/168560671-bacaf039-b25e-4a00-a2ff-65a05fd1f4a4.png)

and below in the class you are only using property (one with `get` and `set`), not a field (one which is starting from lower char - `overrideRules`). This means there is no need in inner field.

Reply:
You can check it out here - [sharplabio](https://sharplab.io/#v2:EYLgtghglgdgPgAQEwEYCwAoBBmABM3AYV0wG9NdL88EUAGXAOQjAFNdSBzVgFwG4Azrz4BfTCKA)

Reply:
Thank you for the links and explanation!